internal/serviceconfig/serviceconfig.go

```diff
@@ -149,6 +149,8 @@ type MethodConfig struct {
 	MaxRespSize *int
 	// RetryPolicy configures retry options for the method.
 	RetryPolicy *RetryPolicy
+
+	Host string
```

Comment:
I don't think we'll want to expose this in the service config. Have you considered using the `iresolver.RPCConfig` context to pass this information along instead?